Tests/SpeziLLMTests/LLMOpenAIParameterTests+Array.swift

```diff
@@ -75,32 +75,40 @@ final class LLMOpenAIParameterArrayTests: XCTestCase {
         
         // Validate parameter schema
         let schemaArrayInt = try XCTUnwrap(llmFunction.schemaValueCollectors["intArrayParameter"])
-        XCTAssertEqual(schemaArrayInt.schema.type, .array)
-        XCTAssertEqual(schemaArrayInt.schema.description, "Int Array Parameter")
-        XCTAssertEqual(schemaArrayInt.schema.minItems, 1)
-        XCTAssertEqual(schemaArrayInt.schema.maxItems, 9)
-        XCTAssertTrue(schemaArrayInt.schema.uniqueItems ?? false)
-        XCTAssertEqual(schemaArrayInt.schema.items?.type, .integer)
+        var schema = schemaArrayInt.schema.value
+        var items = schema["items"] as? [String: Any]
+        XCTAssertEqual(schema["type"] as? String, "array")
+        XCTAssertEqual(schema["description"] as? String, "Int Array Parameter")
+        XCTAssertEqual(schema["minItems"] as? Int, 1)
+        XCTAssertEqual(schema["maxItems"] as? Int, 9)
+        XCTAssertTrue(schema["uniqueItems"] as? Bool ?? false)
+        XCTAssertEqual(items?["type"] as? String, "integer")
         
         let schemaArrayDouble = try XCTUnwrap(llmFunction.schemaValueCollectors["doubleArrayParameter"])
-        XCTAssertEqual(schemaArrayDouble.schema.type, .array)
-        XCTAssertEqual(schemaArrayDouble.schema.description, "Double Array Parameter")
-        XCTAssertEqual(schemaArrayDouble.schema.items?.type, .number)
-        XCTAssertEqual(schemaArrayDouble.schema.items?.minimum, 12.3)
-        XCTAssertEqual(schemaArrayDouble.schema.items?.maximum, 45.6)
+        schema = schemaArrayDouble.schema.value
+        items = schema["items"] as? [String: Any]
+        XCTAssertEqual(schema["type"] as? String, "array")
+        XCTAssertEqual(schema["description"] as? String, "Double Array Parameter")
+        XCTAssertEqual(items?["type"] as? String, "number")
+        XCTAssertEqual(items?["minimum"] as? Double, 12.3)
+        XCTAssertEqual(items?["maximum"] as? Double, 45.6)
         
         let schemaArrayBool = try XCTUnwrap(llmFunction.schemaValueCollectors["boolArrayParameter"])
-        XCTAssertEqual(schemaArrayBool.schema.type, .array)
-        XCTAssertEqual(schemaArrayBool.schema.description, "Bool Array Parameter")
-        XCTAssertEqual(schemaArrayBool.schema.items?.type, .boolean)
-        XCTAssertEqual(schemaArrayBool.schema.items?.const, "true")
+        schema = schemaArrayBool.schema.value
+        items = schema["items"] as? [String: Any]
```

Comment:
It's a bit ugly that we loose all underlying type information of the respective properties of the schema here.. Mabye some computed properties on the schema with a specific type would make sense?

Reply:
I agree. The problem is that the underlying type for the schema is the generated `OpenAPIRuntime.OpenAPIObjectContainer` type, which boils down to a `[String: (any Sendable)?]` type and is what we get back from the API / generated calls.

Which `schema` are you referring to here? I don't think I'm following where you would want that computed property to be.